pkg/kubelet/status/status_manager.go

```diff
@@ -455,6 +456,7 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 			deleteOptions.Preconditions = metav1.NewUIDPreconditions(string(pod.UID))
 			if err = m.kubeClient.Core().Pods(pod.Namespace).Delete(pod.Name, deleteOptions); err == nil {
 				glog.V(3).Infof("Pod %q fully terminated and removed from etcd", format.Pod(pod))
+				metrics.PodDeletionLatency.Observe(metrics.SinceInMicroseconds(pod.DeletionTimestamp.Time))
```

Comment:
For example, kubelet records a timestamp as an annotation in the pod object when it first sees the pod. https://github.com/kubernetes/kubernetes/blob/v1.6.2-beta.0/pkg/kubelet/config/config.go#L413
Perhaps you can use the same mechanism.